webgui/src/flipmouse/js/tabslot.js

```diff
@@ -1,13 +1,61 @@
 window.tabSlot = {};
 
+//require is only useful in electron.
+var inElectron = false;
+var activateOnProcess;
+
+//check every second, if a process is active, which should load a slot.
+setInterval(function() {
+	if (typeof activateOnProcess !== 'undefined') {
+		//load list of currently running processes
+		ps.lookup({}, function(err, resultList ) {
+			if (err) {
+				throw new Error( err );
+			}
+			//we just need process names
+			resultList = resultList.map(a => a.command);
+			//remove duplicates (we trigger on one process name anyway and the list is shorter for iter)
+			resultList = resultList.filter((v, i, a) => a.indexOf(v) === i);
+			//compare list of slots/process names against currently running ones
+			for (var i = 0; i < activateOnProcess.length; i++) {
+				var index = resultList.findIndex(function(item, j){
+					return item === activateOnProcess[i]["process"];
+				});
+				if(index !== -1)
+				{
+					//load this slot, but only if not already active...
+					if(flip.getCurrentSlot() !== activateOnProcess[i]["slot"])
+					{
+						var config = flip.setSlot(activateOnProcess[i]["slot"]);
+						L('.slot-select').forEach(function (elem) {
+							elem.value = activateOnProcess[i]["slot"];
+						});
+						tabAction.init();
```

Comment:
why this `tabAction.init()` is needed?

Reply:
Is it needed for updating the actions tab after a slot switch?

Reply:
hmm, yes, possibly. but I wonder if it's needed or if `tabAction.init()` isn't called anyway on opening the action tab.

Reply:
It is necessary for one reason:
If the tab "actions" is active, and a slot switch takes place via the window focus, this call updates the actions tab with the currently active settings.
If it is commented out, the tab does not update.

Reply:
ah, ok, makes sense 👍